src/components/submission-common.js

```diff
@@ -107,25 +107,29 @@ export const resolveWdsApp = apps => {
   // look explicitly for a RUNNING app named 'wds-${app.workspaceId}' -- if WDS is healthy and running, there should only be one app RUNNING
   // an app may be in the 'PROVISIONING', 'STOPPED', 'STOPPING', which can still be deemed as an OK state for WDS
   const healthyStates = ['RUNNING', 'PROVISIONING', 'STOPPED', 'STOPPING']
-  const namedApp = apps.filter(app => app.appType === getConfig().wdsAppTypeName && app.appName === `wds-${app.workspaceId}` && healthyStates.includes(app.status))
-  if (namedApp.length === 1) {
-    return namedApp[0]
-  }
 
-  //Failed to find an app with the proper name, look for a RUNNING WDS app
-  const runningWdsApps = apps.filter(app => app.appType === getConfig().wdsAppTypeName && app.status === 'RUNNING')
-  if (runningWdsApps.length > 0) {
-    // Evaluate the earliest-created WDS app
-    runningWdsApps.sort((a, b) => new Date(a.auditInfo.createdDate).valueOf() - new Date(b.auditInfo.createdDate).valueOf())
-    return runningWdsApps[0]
-  }
+  // WDS appType is checked first and takes precedence over CROMWELL apps in the workspace
+  for (const appTypeName of getConfig().wdsAppTypeName) {
```

Comment:
small nit: Since `wdsAppTypeName` is an array maybe we can rename it to `wdsAppTypeNames`? But I don't have a strong opinion on this either way.

Reply:
I second this. I actually think it's kind of important!

Reply:
I think I remember us talking about some logic that is being borrowed from elsewhere (maybe terra-ui). Is that logic? Just curious

Reply:
Should this be renamed to something like `appTypeNames` since there will be multiple apps?